test/gui/shared/scripts/pageObjects/SyncConnection.py

```diff
@@ -142,3 +151,16 @@ def confirm_folder_sync_connection_removal():
         squish.clickButton(
             squish.waitForObject(SyncConnection.REMOVE_FOLDER_SYNC_CONNECTION_BUTTON)
         )
+
+
+    @staticmethod
+    def get_permission_error_message():
+        # Wait for permission error label to appear
+        result = squish.waitFor(
+            lambda: object.exists(SyncConnection.PERMISSION_ERROR_LABEL),
+            SyncConnection.PERMISSION_ERROR_WAIT_TIMEOUT * 1000
+        )
+        if result:
+            return str(squish.waitForObject(SyncConnection.PERMISSION_ERROR_LABEL, get_config("lowestSyncTimeout") * 1000).text)
+        else:
+            return ""
```

Comment:
if we add `wait_for_error_label` method then we can directly return like this. this will throw if the label is not found
```suggestion
        return str(squish.waitForObject(SyncConnection.PERMISSION_ERROR_LABEL).text)
```